### DISCUSSION TITLE
AutoKeras for 2d structured data

### DISCUSSION THREAD

**Discussion by yunoshev** [keras-team/autokeras#1361] [Q&A]
Hi, I have the following dataset shape - (4213, 1024, 56) (4213, 8), so I have some 4213 records with shape 1024x56 and need to predict the class of each record. How I should prepare data for StructuredDataClassifier? Thanks.

**Comment by haifeng-jin** [comment]
I assume your data is not really structured data.
You should use the functional API to build your custom search space using [Input](https://autokeras.com/node/#input) and follow this [tutorial](https://autokeras.com/tutorial/customized/#customized-search-space).
